esi_leap/objects/offer.py

```diff
@@ -88,3 +113,25 @@ def expire(self, context=None):
         # expire offer
         self.status = statuses.EXPIRED
         self.save(context)
+
+    def to_marketplace_dict(self):
+        # modifiy the offer data in order to be compatible with flocx-market POST offer api
+        # change fields name
+        offer_dict = self.to_dict()
+        offer_dict['start_time'] = offer_dict.pop('start_date').isoformat()
+        offer_dict['end_time'] = offer_dict.pop('end_date').isoformat()
+        offer_dict['server_config'] = offer_dict.pop('properties')
+        offer_dict['server_id'] = offer_dict.pop('resource_uuid')
+        offer_dict['provider_id'] = offer_dict.pop('uuid')
+        # remove unnecessary feilds
+        offer_dict.pop('created_at')
+        offer_dict.pop('updated_at')
+        offer_dict.pop('id')
+        offer_dict.pop('project_id')
+        offer_dict.pop('resource_type')
+        # fake fields
+        offer_dict['creator_id'] = 'Alice1992'
+        offer_dict['marketplace_date_created'] = '2016-07-16T19:20:30'
```

Comment:
Don't pass this value in; in theory flocx_market should default this value

Reply:
@fvukelic I think marketplace should have a default 'marketplace_date_created' value so that provider doesn't need to send this value to flocx_market. Right now I just fake this value in order to avoid 500 error in flocx_market when sending a post request.

Reply:
Ah - maybe use the current date here for now then.

Reply:
Besides, the offer model in provider doesn't have a cost field, so I fake a cost value and pass it to marketplace.